src/init.cpp

```diff
@@ -954,10 +954,10 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
     if (mapArgs.count("-minrelaytxfee"))
     {
         CAmount n = 0;
-        if (ParseMoney(mapArgs["-minrelaytxfee"], n) && n > 0)
-            ::minRelayTxFee = CFeeRate(n);
-        else
+        if (!ParseMoney(mapArgs["-minrelaytxfee"], n))
             return InitError(AmountErrMsg("minrelaytxfee", mapArgs["-minrelaytxfee"]));
+        // High fee check is done afterward in CWallet::ParameterInteraction()
+        ::minRelayTxFee = CFeeRate(n);
```

Comment:
We don't do the `n>0` elsewhere because it is redundant:
```
src/test/util_tests.cpp:    BOOST_CHECK(!ParseMoney("-1", ret));
```

Also, the syntax is preferable:

```cpp
if(fail)
  return;
doStuff();
```

vs

```cpp
if(!fail)
  doStuff();
else
  return;